packages/server-wallet/src/db/migrations/20201014174039_objectives.ts

```diff
@@ -1,25 +1,27 @@
 import * as Knex from 'knex';
 
 const tableName = 'objectives';
+const associativeTableName = 'objectives_channels';
 
 export async function up(knex: Knex): Promise<any> {
-  return knex.schema.createTable(tableName, function(table) {
+  await knex.schema.createTable(tableName, function(table) {
     table.string('objective_id').primary();
     table.string('status').notNullable();
     table.string('type').notNullable();
-    table.string('target_channel_id');
-    table.string('funding_strategy');
-    table.string('ledger_id');
-    table.string('joint_channel_id');
-    table.string('guarantor_id');
+    table.jsonb('data').notNullable();
```

Comment:
Hopefully this presents a lower risk to future changes in our objective types.

Reply:
I would prefer a data model with fewer foot guns, but I think this is an entirely reasonable choice given our current timeline. (I might go further and say this is our best choice for the moment.)